### PR TITLE
Add EqualsTo() method to TcoObject

### PR DESCRIPTION
**Pull request by PTKu**
Implemenation of ```EqualsTo``` method #34.

I have chose ```EqualsTo``` instead of ```Equals``` due to possible confusion between .net ```object.Equals(object)``` and ```TcoObject.Equals(ITcoObject)```.

- [x] In default implementation returns true when object references equals
- [x] In an overridden method values should be compared for equality

Example of ```EqualsTo``` override

~~~ iecst
METHOD PUBLIC EqualsTo : BOOL
VAR_INPUT
	Object : ITcoObject;
END_VAR

VAR
	_object : POINTER TO TcoObjectEqualsTest;
END_VAR
//-------------------------------------------------------
IF(__QUERYPOINTER(Object,  _object)) THEN
	EqualsTo := _object^._SomeNumber = THIS^._SomeNumber AND _object^._SomeString = THIS^._SomeString;
END_IF;	
~~~ 

+ Tests

### CODE REVIEW

File: src/TcoCore/src/XaeTcoCore/TcoCore/POUs/Prototypes/TcoObject/TcoObject.TcPOU [hunk 1]
```diff
@@ -108,6 +108,50 @@ END_VAR
...
+					EqualsTo := _object^._SomeNumber = THIS^._SomeNumber AND _object^._SomeString = THIS^._SomeString;
+				END_IF;	
+			</code>
+		</example>
+	</docu>	
+~*)
+METHOD PUBLIC EqualsTo : BOOL
+VAR_INPUT
+	Object : ITcoObject;
+END_VAR
```
Comment (PTKu) [inline src/TcoCore/src/XaeTcoCore/TcoCore/POUs/Prototypes/TcoObject/TcoObject.TcPOU:144]: Here we may have an issue when when we create objects dynamically at run-time. The address ```indentity``` is assigned at construction and never changes over the course of the run. Dynamically created objects may match identity of existing if there were online changes performed at some point between the start up and dynamic allocation. Something to discuss if we consider it an issue we can adopt this approach https://github.com/Inxton/TcOpen/issues/34#issuecomment-816643365.

File: src/TcoCore/src/XaeTcoCore/TcoCoreTests/Tests/TcoObject/TcoObjectTest.TcPOU [hunk 2]
```diff
@@ -24,6 +27,41 @@ METHOD INTERNAL CallTaskInstancies]]></Declaration>
...
+VAR 
+	referenceTo_TcoStateTest_A : REFERENCE TO TcoStateTest;
+END_VAR]]></Declaration>
+      <Implementation>
+        <ST><![CDATA[CASE _case OF
+	0:
+		referenceTo_TcoStateTest_A REF= _TcoStateTest_A;
+		EqualsTest := _TcoStateTest_A.EqualsTo(referenceTo_TcoStateTest_A);
+    1:	
+		EqualsTest := _TcoStateTest_A.EqualsTo(_TcoStateTest_B);
```
Comment (GISED-Link) [inline src/TcoCore/src/XaeTcoCore/TcoCoreTests/Tests/TcoObject/TcoObjectTest.TcPOU:58]: `EqualsTest := _TcoStateTest_B.EqualsTo(_TcoStateTest_A);` should also be tested... ?

File: src/TcoCore/src/XaeTcoCore/TcoCoreTests/Tests/TcoObject/TcoObjectTest.TcPOU [hunk 3]
```diff
@@ -24,6 +27,41 @@ METHOD INTERNAL CallTaskInstancies]]></Declaration>
...
+
+VAR 
+	referenceTo_TcoStateTest_A : REFERENCE TO TcoStateTest;
+END_VAR]]></Declaration>
+      <Implementation>
+        <ST><![CDATA[CASE _case OF
+	0:
+		referenceTo_TcoStateTest_A REF= _TcoStateTest_A;
+		EqualsTest := _TcoStateTest_A.EqualsTo(referenceTo_TcoStateTest_A);
+    1:
```
Comment (GISED-Link) [inline src/TcoCore/src/XaeTcoCore/TcoCoreTests/Tests/TcoObject/TcoObjectTest.TcPOU:57]: The use of a formatter will be wellcome. Here we are messing between space and tab in the begining of the lines

File: src/TcoCore/src/XaeTcoCore/TcoCore/POUs/Prototypes/TcoObject/TcoObject.TcPOU [hunk 4]
```diff
@@ -108,6 +108,50 @@ END_VAR
...
     </Property>
+    <Method Name="EqualsTo" Id="{4cbabb6f-bb30-0500-2ad1-c579fa67b5bc}">
+      <Declaration><![CDATA[(*~
+	<docu>
+		<summary>
+			Compares this object to another object and returns ```true``` when the reference equals. 
+			When overriden in derived class custom comparison of two objects of the same type can be evaluated.						
+ 		</summary>			
+		<note>
+			Default implementation compares if two objects by reference. If you want you compare actual values of the objects
```
Comment (GISED-Link) [inline src/TcoCore/src/XaeTcoCore/TcoCore/POUs/Prototypes/TcoObject/TcoObject.TcPOU:119]: Default implementation compares only if two objects share the same reference. If you want you compare actual values of the objects you will need to override ```EqualsTo``` method where you evaluate the values for equality.

**Review by GISED-Link — COMMENTED** [review]
Hi, good job, do what is needed. Next time you should add someone to the review, it is always helpfull 😃

**Comment by GISED-Link** [comment]
> I have chose `EqualsTo` instead of `Equals` due to possible confusion between .net `object.Equals(object)` and `TcoObject.Equals(ITcoObject)`.

Good idea

File: src/TcoCore/src/XaeTcoCore/TcoCoreTests/Tests/TcoObject/TcoObjectTest.TcPOU (hunk 3 above)
Comment (PTKu) [reply src/TcoCore/src/XaeTcoCore/TcoCoreTests/Tests/TcoObject/TcoObjectTest.TcPOU:57]: I actually have STweep installed... have to get more familiar with it :)

**Comment by PTKu** [comment]
Thanks @GISED-Link, will incorporate your comments. Sending you invite to this repo otherwise cannot add you as reviewer.

**Comment by PTKu** [comment]
@GISED-Link we are moving from here to TcOpenGroup organization 
with this PR: https://github.com/TcOpenGroup/TcOpen/pull/13
See you there

**Comment by GISED-Link** [comment]
I feel like a VIP :smile:
Thank you for the notification.

I was in a training for the SuperTrak (XTS system but from B&R) system this week so I was not very there for reviews. I have created a cool library to manage all the shuttles. I will add TcUnits tests (better too late than never) and TcOpen so I will have a better undestanding of the project.

**Comment by GISED-Link** [comment]
I feel like a VIP :smile:
Thank you for the notification.

I was in a training for the SuperTrak (XTS system but from B&R) system this week so I was not very there for reviews. I have created a cool library to manage all the shuttles. I will add TcUnits tests (better too late than never) and TcOpen so I will have a better undestanding of the project.

**Pull request merged**